clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -4384,4 +4349,124 @@ HeartbeatMonitoringService getHeartbeatMonitoringService() {
     return heartbeatMonitoringService;
   }
 
+  protected String getDataRecoverySourcePubSub(Version version) {
+    String configuredSource = version.getDataRecoveryVersionConfig().getDataRecoverySourceFabric();
+    int pubSubId = this.serverConfig.getKafkaClusterAliasToIdMap().getInt(configuredSource);
+    String dataRecoverySourcePubSub = this.serverConfig.getKafkaClusterIdToUrlMap().get(pubSubId);
+    if (dataRecoverySourcePubSub == null) {
+      throw new VeniceException(
+          "Unable to get data recovery source pub sub url from the provided config:" + configuredSource);
+    }
+    return dataRecoverySourcePubSub;
+  }
+
+  @Override
+  protected void refreshIngestionContextIfChanged(Store store) throws InterruptedException {
+    super.refreshIngestionContextIfChanged(store);
+
+    Version version = store.getVersion(this.versionNumber);
+    if (version == null) {
+      // Should we fail instead, or trigger a close? It's kind of weird if the version of interest doesn't exist...
+      return;
+    }
+
+    configureNativeReplicationAndDataRecovery(version, false);
+  }
+
+  /**
+   * There are several configurations relating to native replication which are initialized at construction-time but
+   * which can also change later on during run-time. These configurations are impacted by whether the ingestion task
+   * runs in data recovery mode. In order to avoid duplicating the relevant business logic across both the constructor
+   * and the change listener, we consolidate it here. The function logs at most one line, summarizing what it did.
+   *
+   * @param version config coming from ZK
+   * @param constructionTime if true, all fields are guaranteed to be initialized (and logged if DR is enabled),
+   *                         if false, fields will only be mutated and logged if they differ from the intended state
+   */
+  private void configureNativeReplicationAndDataRecovery(Version version, boolean constructionTime) {
+    boolean versionHasDataRecoveryConfig = version.getDataRecoveryVersionConfig() != null;
```

Comment:
It seems like Da Vinci client case is ignored in the new changes - From what was documented in the original Javadoc, for DVC, we should always point the dataRecovery to false.